### PR TITLE
AlertHelper class logic

### PR DESCRIPTION
**Pull request by LeeVoi**
Called the method inside the LoginController to display the alert if logged in as an Admin or ProjectManager.

### CODE REVIEW

File: src/main/java/dk/easv/gui/controllers/LoginController.java
Comment (LordStacker) [inline src/main/java/dk/easv/gui/controllers/LoginController.java:1]: remove alerts from controllers.

**Review by LordStacker — DISMISSED** [review]

**Comment by LeeVoi** [comment]
removed the alert from the controllers

**Comment by LeeVoi** [comment]
finished the task of displaying the documents that are created 48 months ago or longer, the method will compare the date_created in the database and local date/time than will store the names of the documents in a list and display the names with the alert.
Also added a condition that if there is no documents to display the alert will not show.

File: src/main/java/dk/easv/helpers/AlertHelper.java
Comment (patrikvalentiny) [inline src/main/java/dk/easv/helpers/AlertHelper.java:1]: Why is the check performed within an alert helper class in a method that is supposed to show the alert? refactor this to be checked in an appropriate layer and time

File: src/main/java/dk/easv/helpers/AlertHelper.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import java.time.LocalDateTime;
+import java.time.temporal.ChronoUnit;
+import java.util.List;
+import java.util.concurrent.ConcurrentMap;
+import java.util.stream.Collectors;
+
+public class AlertHelper {
+    public static Alert alert;
+    public static void showDefaultAlert(String content, Alert.AlertType type) throws SQLException {
+        LocalDateTime currentDate = LocalDateTime.now();
```
Comment (patrikvalentiny) [inline src/main/java/dk/easv/helpers/AlertHelper.java]: since this is only used once you can call the method straight

**Review by patrikvalentiny — CHANGES_REQUESTED** [review]

File: src/main/java/dk/easv/helpers/AlertHelper.java
Comment (LeeVoi) [reply src/main/java/dk/easv/helpers/AlertHelper.java:1]: moved the method from the AlertHelper class

File: src/main/java/dk/easv/helpers/AlertHelper.java (hunk 1 above)
Comment (LeeVoi) [reply src/main/java/dk/easv/helpers/AlertHelper.java]: called the alert inside controllers AdminView and ProjectManagerView

File: src/main/java/dk/easv/helpers/DocumentHelper.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
+package dk.easv.helpers;
+import dk.easv.be.Document;
+import dk.easv.bll.DocumentLogic;
+
+import java.sql.SQLException;
+import java.util.List;
+
+
+public class DocumentHelper {
+    public static StringBuilder convertToString() throws SQLException {
```
Comment (patrikvalentiny) [inline src/main/java/dk/easv/helpers/DocumentHelper.java]: return string, add the list as an argument use this only for translation to string do not fetch the list from here

File: src/main/java/dk/easv/bll/DocumentLogic.java [hunk 3]
```diff
@@ -27,17 +27,16 @@ public void addImage(int documentId, int index, Image image) throws SQLException
     public List<Content> loadContent(int documentId) throws SQLException {
         return contentDAO.loadContent(documentId);
     }
-    public List<String> showDocumentName() throws SQLException {
+    public static List<Document> showDocumentName() throws SQLException {
```
Comment (patrikvalentiny) [inline src/main/java/dk/easv/bll/DocumentLogic.java]: rename this method to something more appropriate remove static reference

File: src/main/java/dk/easv/gui/controllers/AdminViewController.java [hunk 4]
```diff
@@ -41,13 +42,10 @@ public void initialize(URL location, ResourceBundle resources) {
...
 
-            DocumentLogic documentLogic = new DocumentLogic();
-            List<String> documentNames = documentLogic.showDocumentName();
-            StringBuilder docNames = new StringBuilder("The following documents are 48 months old:\n");
-            for (String documentName : documentNames) {
-                docNames.append(documentName).append("\n");
-            }
-            AlertHelper.showDefaultAlert(docNames.toString(), Alert.AlertType.CONFIRMATION);
+            DocumentHelper documentHelper = new DocumentHelper();
+            StringBuilder documentNames = documentHelper.convertToString();
```
Comment (patrikvalentiny) [inline src/main/java/dk/easv/gui/controllers/AdminViewController.java]: make the method convertToString() static and return String, you can call it inline with the olddocuments parameter

File: src/main/java/dk/easv/gui/controllers/AdminViewController.java [hunk 5]
```diff
@@ -41,13 +42,10 @@ public void initialize(URL location, ResourceBundle resources) {
...
+
+            AlertHelper.showDefaultAlert(documentNames.toString(), Alert.AlertType.CONFIRMATION);
```
Comment (patrikvalentiny) [inline src/main/java/dk/easv/gui/controllers/AdminViewController.java]: call it inline here get the documents list straight by calling getOldDocuments from model

File: src/main/java/dk/easv/gui/controllers/ProjectManagerViewController.java [hunk 6]
```diff
@@ -67,15 +69,12 @@ public class ProjectManagerViewController extends RootController {
...
-            List<String> documentNames = documentLogic.showDocumentName();
-            StringBuilder docNames = new StringBuilder("The following documents are 48 months old:\n");
-            for (String documentName : documentNames) {
-                docNames.append(documentName).append("\n");
-            }
-            AlertHelper.showDefaultAlert(docNames.toString(), Alert.AlertType.CONFIRMATION);
+            DocumentHelper documentHelper = new DocumentHelper();
+            StringBuilder documentNames = documentHelper.convertToString();
+
+            AlertHelper.showDefaultAlert(documentNames.toString(), Alert.AlertType.CONFIRMATION);
```
Comment (patrikvalentiny) [inline src/main/java/dk/easv/gui/controllers/ProjectManagerViewController.java]: same as previous controller

File: src/main/java/dk/easv/helpers/DocumentHelper.java [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+        List<Document> oldDocuments = DocumentLogic.showDocumentName();
```
Comment (patrikvalentiny) [inline src/main/java/dk/easv/helpers/DocumentHelper.java]: remove this use argument

File: src/main/java/dk/easv/helpers/DocumentHelper.java [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+        List<String> documentNames = oldDocuments.stream()
```
Comment (patrikvalentiny) [inline src/main/java/dk/easv/helpers/DocumentHelper.java]: you are looping through the list twice for no reason, beware that you have to understand the code you write for the exam so steer clear of just copying stuff you dont understand

File: src/main/java/dk/easv/helpers/DocumentHelper.java [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+                .map(Document::getName)
+                .toList();
+
+        StringBuilder docNames = new StringBuilder("The following documents are 48 months old:\n");
+        for (String documentName : documentNames) {
```
Comment (patrikvalentiny) [inline src/main/java/dk/easv/helpers/DocumentHelper.java]: this is the second time you loop through the list. you can just loop through the olddocuments list

File: src/main/java/dk/easv/helpers/DocumentHelper.java [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+            docNames.append(documentName).append("\n");
```
Comment (patrikvalentiny) [inline src/main/java/dk/easv/helpers/DocumentHelper.java]: in here call .getName() on the current document you are looping through

**Review by patrikvalentiny — CHANGES_REQUESTED** [review]

File: src/main/java/dk/easv/helpers/DocumentHelper.java (hunk 2 above)
Comment (LeeVoi) [reply src/main/java/dk/easv/helpers/DocumentHelper.java]: adding the List as an argument is not working, as the List is returned as NULL somehow and it's giving an error when logging in

File: src/main/java/dk/easv/gui/controllers/ProjectManagerViewController.java [hunk 11]
```diff
@@ -71,9 +72,9 @@ public class ProjectManagerViewController extends RootController {
     @Override
     public void initialize(URL location, ResourceBundle resources) {;
         try {
-            StringBuilder documentNames = DocumentHelper.convertToString();
-
-            AlertHelper.showDefaultAlert(documentNames.toString(), Alert.AlertType.CONFIRMATION);
+            DocumentLogic documentLogic = new DocumentLogic();
+            List<Document> oldDocuments = documentLogic.showOldDocuments();
```
Comment (patrikvalentiny) [inline src/main/java/dk/easv/gui/controllers/ProjectManagerViewController.java]: pass this through model

File: src/main/java/dk/easv/gui/controllers/AdminViewController.java [hunk 12]
```diff
@@ -41,9 +41,10 @@ public class AdminViewController extends RootController {
     public void initialize(URL location, ResourceBundle resources) {
         try {
             documentModel = new DocumentModel();
-            StringBuilder documentNames = DocumentHelper.convertToString();
 
-            AlertHelper.showDefaultAlert(documentNames.toString(), Alert.AlertType.CONFIRMATION);
+            DocumentLogic documentLogic = new DocumentLogic();
+            List<Document> oldDocuments = documentLogic.showOldDocuments();
```
Comment (patrikvalentiny) [inline src/main/java/dk/easv/gui/controllers/AdminViewController.java]: pass this through model

File: src/main/java/dk/easv/gui/controllers/AdminViewController.java [hunk 13]
```diff
@@ -41,9 +41,10 @@ public class AdminViewController extends RootController {
     public void initialize(URL location, ResourceBundle resources) {
         try {
             documentModel = new DocumentModel();
-            StringBuilder documentNames = DocumentHelper.convertToString();
 
-            AlertHelper.showDefaultAlert(documentNames.toString(), Alert.AlertType.CONFIRMATION);
+            DocumentLogic documentLogic = new DocumentLogic();
```
Comment (patrikvalentiny) [inline src/main/java/dk/easv/gui/controllers/AdminViewController.java]: remove

File: src/main/java/dk/easv/helpers/DocumentHelper.java [hunk 14]
```diff
@@ -1,22 +1,18 @@
...
-    public static StringBuilder convertToString() throws SQLException {
-        List<Document> oldDocuments = DocumentLogic.showDocumentName();
-        List<String> documentNames = oldDocuments.stream()
-                .map(Document::getName)
-                .toList();
-
-        StringBuilder docNames = new StringBuilder("The following documents are 48 months old:\n");
-        for (String documentName : documentNames) {
-            docNames.append(documentName).append("\n");
+    public static String convertToString(List<Document> oldDocuments) throws SQLException {
```
Comment (patrikvalentiny) [inline src/main/java/dk/easv/helpers/DocumentHelper.java]: remove throws statement

File: src/main/java/dk/easv/gui/controllers/ProjectManagerViewController.java [hunk 15]
```diff
@@ -71,9 +72,9 @@ public class ProjectManagerViewController extends RootController {
     @Override
     public void initialize(URL location, ResourceBundle resources) {;
         try {
-            StringBuilder documentNames = DocumentHelper.convertToString();
-
-            AlertHelper.showDefaultAlert(documentNames.toString(), Alert.AlertType.CONFIRMATION);
+            DocumentLogic documentLogic = new DocumentLogic();
```
Comment (patrikvalentiny) [inline src/main/java/dk/easv/gui/controllers/ProjectManagerViewController.java]: remove

**Review by patrikvalentiny — CHANGES_REQUESTED** [review]

**Review by patrikvalentiny — APPROVED** [review]

**Pull request merged**